src/main/java/Refactoring.java

```diff
@@ -1,7 +1,43 @@
+import java.util.ArrayList;
+import java.util.List;
+
 import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.ParenthesizedExpression;
+import org.eclipse.jdt.core.dom.PrefixExpression;
 import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
 
 public abstract class Refactoring {
 	public abstract boolean isApplicable(ASTNode node);
+
 	public abstract void apply(ASTNode node, ASTRewrite rewriter);
+
+	/**
+	 * Recursively analyze an expression to get all the subexpressions that comprise
+	 * it. Finds inner expressions of PrefixExpressions, ParenthesizedExpressions,
```

Comment:
This Javadoc comment mixes implementation detail ("how it works") with the goal ("what are we looking for"), because it specifies the goal ("subexpressions that comprise it") in terms of the technical components ("Prefix..."). What is the _purpose_ of this method? I _think_ it's something like "find all the boolean subexpressions in a boolean expression" (given the specialization to `NOT` in the implementation and the fact that postfix expressions are not considered), but the Javadoc doesn't say so.